entities/JsonHandlers/Serializers/EntityJsonSerializer.cpp

```diff
@@ -4,43 +4,36 @@
 #include "../../POI.h"
 #include "../../Junction.h"
 #include "../../Way.h"
+#include "../../geometry/PointList.h"
 
 rapidjson::Value EntityJsonSerializer::entityToJson(Entity *entity, rapidjson::Document::AllocatorType &allocator) {
     rapidjson::Value json;
     json.SetObject();
+    json = setString(std::move(json), "type", entity->getType(), allocator);
     json = setString(std::move(json), "id", entity->getId(), allocator);
-
     json = setString(std::move(json), "name", entity->getName(), allocator);
-
     if (entity->getDescription()) {
         json = setString(std::move(json), "description", entity->getDescription().value(), allocator);
     }
-
     json = setStringVector(std::move(json), "category_tags", entity->getCategoryTags(), allocator);
-
     return json;
 }
 
 rapidjson::Value EntityJsonSerializer::toJson(POI *entity, rapidjson::Document::AllocatorType &allocator) {
-    rapidjson::Value json = entityToJson(entity, allocator);
-    json = setStringVector(std::move(json), "accessibility", entity->getAccessibility(), allocator);
-    json = setGeometry(std::move(json), entity->getGeometry().get(),
-                       allocator);
-
-    return json;
-}
-
-rapidjson::Value EntityJsonSerializer::setGeometry(rapidjson::Value entityJson, Geometry *geometry,
-                                                   rapidjson::Document::AllocatorType &allocator) {
+    rapidjson::Value entityJson = entityToJson(entity, allocator);
+    entityJson = setStringVector(std::move(entityJson), "accessibility", entity->getAccessibility(), allocator);
+    Geometry *geometry = entity->getGeometry().get();
     rapidjson::Value geometryJson = geometry->toJson(allocator);
     entityJson.AddMember("geometry", geometryJson, allocator);
+
     return entityJson;
 }
 
 rapidjson::Value EntityJsonSerializer::toJson(Junction *entity, rapidjson::Document::AllocatorType &allocator) {
     rapidjson::Value json = entityToJson(entity, allocator);
-    rapidjson::Value coord = coordinatesJsonSerializer.toJson(entity->getCoordinates(), allocator);
-    json.AddMember("coordinates", coord, allocator);
+    Geometry *geometry = entity->getGeometry().get();
```

Comment:
It's generally better to not use get().
I used it elsewhere because I needed to send objects as function parameters and use overloading.